WatsonWebserver/Parameters/RouteManager.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace WatsonWebserver
+{
+    public class RouteManager
+    {
+        Dictionary<string, ApiControllerMethodInfo> routes = new Dictionary<string, ApiControllerMethodInfo>();
+
+        internal void AddRoute(string routePrefix, string methodName, Type classType, MethodInfo classMethod, HttpMethod httpMethod)
+        {
+            routes.Add($"/{routePrefix.Trim('/')}/{methodName.Trim('/')}", 
+                new ApiControllerMethodInfo(classType, classMethod, httpMethod));
+        }
+
+        internal ApiControllerMethodInfo Match(HttpRequest req)
+        {
+            routes.TryGetValue(req.Url.RawWithoutQuery, out var invokeInfo);
```

Comment:
I'm thinking about amending this so that the key includes the HTTP method, that way, there wouldn't need to be separate attributes per verb (HttpGet, HttpPost, etc) but rather a consolidated attribute like ```AttributeRoute(HttpMethod.GET)```.  Thoughts?